Tests/ConsulServiceDiscoveryTests/ConsulTests.swift

```diff
@@ -26,6 +26,43 @@ final class ConsulTests: XCTestCase {
         try consul.syncShutdown()
     }
 
+    func testHealthCheckError() async throws {
+        let consul = Consul()
+
+        let serviceName = "test_service"
+        let processInfo = ProcessInfo.processInfo
+        let serviceID = "\(processInfo.hostName)-\(processInfo.processIdentifier)-\(serviceName)-\(#line)"
+
+        let checkID = "service:\(serviceID)"
+        let future = consul.agent.check(checkID, status: .passing)
+
+        var continuation: AsyncStream<Result<Void, Error>>.Continuation?
```

Comment:
```swift
let (stream, continuation) = AsyncStream.makeStream(of: Result<Void, Error>.self)
```

Reply:
It will require to add package-concurrency-helpers as a dependency, what I would not like to do for 2 lines of code.

Reply:
Nope, it's in standard Swift library since 5.9 -- <https://developer.apple.com/documentation/swift/asyncstream/makestream(of:bufferingpolicy:)> [](edited-with-Axolo)